lib/viewcomponents/io.jsx

```diff
@@ -0,0 +1,66 @@
+var React = require('react');
+var mouseDownDrag = require('../utilities/mouseDownDrag');
+var LabelPosition = require('../enum/ioLabelPosition');
+
+var ioRadius = 4;
+/*
+props:
+- isInput {Boolean}
+- label {String}
+- x {Number}
+- y {Number}
+- size (radius) {Number}
+- labelPosition
+ */
+var IO = React.createClass({
+    /* it's the responsibility of the Node to position the IO since it's in its coordinate space */
+    render: function() {
+        var isInput = this.props.isInput;
+        var ioClass = isInput ? 'input' : 'output';
+
+        var labelPosition = this._getLabelPositioningData(this.props.labelPosition, this.props.size || ioRadius);
+
+        return (
+           <g
+            className={'io ' + ioClass}
```

Comment:
There's a react addon that handles className concatenation. If we get any classNames much more complicated than this, I would recommend that we use the addon.